python-client/giskard/models/base/wrapper.py

```diff
@@ -99,6 +102,12 @@ def __init__(
     def _preprocess(self, data):
         if self.data_preprocessing_function:
             return self.data_preprocessing_function(data)
+
+        # In the case where a dataframe has one column, most sklearn models require a Series instead of a DataFrame.
+        # It could also be the case of PredictionFunctionModel that calls sklearn models.
+        if self.accepts_only_pd_series:
+            return data[data.columns[0]]
```

Comment:
Maybe should we check the column name still ?

Reply:
Shouldn'it failed if you train with a df with columns a, y (y being the target) and then you try to predict with a colname 'y', or 'z' ?
I think your df can have more than 1 col, but the used col is only one, it should still be the proper one no ?

Reply:
Or is it done before ?

Reply:
this one is taken care of ✅ Thanks!